src/khepri_machine.erl

```diff
@@ -1207,27 +1215,29 @@ apply(
   Meta,
   #register_projection{pattern = PathPattern, projection = Projection},
   State) ->
-    Tree = get_tree(State),
+    ProjectionName = Projection#khepri_projection.name,
     ProjectionTree = get_projections(State),
-    Reply = khepri_projection:init(Projection),
-    State1 = case Reply of
-                 ok ->
-                     restore_projection(Projection, Tree, PathPattern),
-                     ProjectionTree1 = khepri_pattern_tree:update(
-                                         ProjectionTree,
-                                         PathPattern,
-                                         fun (?NO_PAYLOAD) ->
-                                                 [Projection];
-                                             (Projections) ->
-                                                 [Projection | Projections]
-                                         end),
-                     erase(compiled_projection_tree),
-                     set_projections(State, ProjectionTree1);
-                 _  ->
-                     State
-             end,
-    Ret = {State1, Reply},
-    bump_applied_command_count(Ret, Meta);
+    case has_projection(ProjectionTree, ProjectionName) of
+        true ->
+            Ret = {State, {error, exists}},
```

Comment:
Could you please use the `?khepri_error(Name, Info)` macro to create the error and give more context?

Probably something like that, to follow other error patterns:
```erlang
Info = #{name => ProjectionName},
Reason = ?khepri_error(projection_already_exists, Info),
{error, Reason}.
```
I'm just not sure about the name `projection_already_exists`.

Reply:
That said, do we want to reject an update of a projection?

Reply:
That sounds good to me. I believe we already handle the `{error, exists}` in the server so we will need to be careful to change that when we update. I would also like for the server to use the new `has_projection` function though rather than relying on idempotency, since submitting the compiled projection functions each time can only inflate Khepri's Ra log.

`projection_already_exists` sounds good to me :+1:

Reply:
> do we want to reject an update of a projection

Currently we do so that registering a projection is idempotent. Doing so makes updates more difficult though. The way to do it currently is `unregister_projection` and then `register_projection`

Reply:
I understand, thank you!